src/hotspot/cpu/aarch64/aarch64_neon_ad.m4

```diff
@@ -1553,55 +1553,50 @@ VFABD(fabd, fabd, 4, F, X, S, 128)
 VFABD(fabd, fabd, 2, D, X, D, 128)
 dnl
 define(`VREPLICATE', `
-instruct replicate$3$4$5`'(vec$6 dst, $7 ifelse($7, immI0, zero, $7, immI, con, src))
-%{
-  predicate(UseSVE == 0 && ifelse($8, `',
-                                  n->as_Vector()->length() == $3,
-                                  (n->as_Vector()->length() == $3 ||`
-                            'n->as_Vector()->length() == $8)));
-  match(Set dst (Replicate`'ifelse($7, immI0, I, $4) ifelse($7, immI0, zero, $7, immI, con, $7, zero, I, src)));
-  ins_cost(INSN_COST);
-  format %{ "$1  $dst, $ifelse($7, immI0, zero, $7, immI, con, src)`\t# vector ('ifelse($4$7, SimmI, $3H, $2, eor, 4I, $3$4)`)"' %}
-  ins_encode %{
-    __ $2(as_FloatRegister($dst$$reg), __ ifelse(
-            $2, eor, T16B, T$3`'$9),ifelse(
-            `$4 $7', `B immI', ` '$con$$constant & 0xff,
-            `$4 $7', `S immI', ` '$con$$constant & 0xffff,
-            `$4 $7', `I immI', ` '$con$$constant,
-            `$2', eor,`
-           as_FloatRegister($dst$$reg),
-           as_FloatRegister($dst$$reg)',
-            `$7', vRegF,`
+instruct replicate$2$3$4`'(vec$5 dst, $6 ifelse($4, _imm, con, src))
+%{
+  predicate(UseSVE == 0 && ifelse($7, `',
+                                  n->as_Vector()->length() == $2,
+                                  (n->as_Vector()->length() == $2 ||`
+                            'n->as_Vector()->length() == $7)));
+  match(Set dst (Replicate$3 ifelse($4, _imm, con, src)));
+  ins_cost(INSN_COST);
+  format %{ "ifelse($1, mov, movi, $1)  $dst, $ifelse($4, _imm, con, src)`\t# vector ('ifelse($3$4, S_imm, $2H, $2$3)`)"' %}
+  ins_encode %{
+    __ $1(as_FloatRegister($dst$$reg), __ T$2$8,ifelse(
+            `$3 $4', `B _imm', ` '$con$$constant & 0xff,
+            `$3 $4', `S _imm', ` '$con$$constant & 0xffff,
+            `$3 $4', `I _imm', ` '$con$$constant,
+            `$3 $4', `L _imm', ` '$con$$constant,
+            `$6', vRegF,`
```

Comment:
This `ifelse` is rather confusing. Can't the '& 0xff' be passed as arguments into `VREPLICATE`?

Reply:
Done. I think these rules should not have been merged by force. It's hard to maintain.

Reply:
Thanks for your suggestion. Fixed.